client/src/views/page-components/account/AccountHistoryModal/AccountHistoryModal.ts

```diff
@@ -0,0 +1,388 @@
+
+      const submitArguments = {
+        //   user:,
+        card: this.payMethod.currentCardName,
+        category: this.choicedCategoryName,
+        money: this.moneyInput.value as HTMLInputElement,
+        date: this.dateInput.value as HTMLInputElement,
+        content: historyContent.value,
+        type: this.type,
+      };
+      console.log('Form Success');
+
+      this.closeModal();
+      handleEvent.fire('createaccounthistory', { state: history.state, submitArguments });
+      // 옵저버 발동
+    } else {
+      this.showAlert('.history-form__confirm-alert');
+    }
+  }
+
+  checkIncomeOrExpenditure() {
+    if (this.choicedCategoryName === '용돈') {
+      this.type = 'income';
+    } else {
+      this.type = 'expenditure';
+      this.moneyInput.value = '-' + this.moneyInput.value;
+    }
+  }
+  onKeyUpHandler(e: KeyboardEvent) {
+    this.onKeyUpDate(e);
+    this.onKeyUpMoney(e);
+  }
+
+  onFocusOutDateInputHandler() {
+    this.onFocusOutDate();
+  }
+
+  onFocusOutMoneyInputHandler() {
+    this.onFocusOutMoney();
+  }
+
+  onFocusOutDate() {
+    const target = $('.history-form__date');
+    const ValidationLengthResult = this.checkDateInputLengthValidation(target, dateInputMaxLength);
+    const ValidationValueResult = this.checkDateInputValueValidation(target);
+
+    if (ValidationLengthResult || ValidationValueResult) {
+      this.showAlert('.date-alert');
+      this.dateValueValidation = false;
+    } else {
+      this.dateValueValidation = true;
+    }
+    this.formatDateValue();
+  }
+
+  formatDateValue() {
+    const DateValue: string = this.dateInput.value;
+    const DateArray = DateValue.split('');
+    // 4, 7
+    if (DateValue.length > 6) {
+      DateArray.splice(4, 0, '.');
+      DateArray.splice(7, 0, '.');
+      this.dateInput.value = DateArray.join('');
+    } else if (DateValue.length > 4) {
+      DateArray.splice(4, 0, '.');
+      this.dateInput.value = DateArray.join('');
+    }
+  }
+
+  onFocusOutMoney() {
+    const target = $('.history-form__money');
+    const ValidationResult = this.checkMoneyInputLengthValidation(target, moneyInputMinLength);
+
+    if (ValidationResult) {
+      this.showAlert('.money-alert');
+      this.moneyValueValidation = false;
+    } else {
+      this.moneyValueValidation = true;
+    }
+    this.formatMoneyValue();
+  }
+
+  formatMoneyValue() {
+    const MoneyValue: string = this.moneyInput.value;
+    const MoneyArray = MoneyValue.split('');
+    const MoneyLength = MoneyValue.length;
+
+    if (MoneyLength > 3) {
+      let count = 1;
+      for (let i = MoneyLength - 1; i > 0; i--) {
+        if (count % 3 === 0) {
+          MoneyArray.splice(i, 0, ',');
+        }
+        count++;
+      }
+    }
+
+    const result = MoneyArray.join('');
+    this.moneyInput.value = result;
+  }
+
+  showAlert(target) {
+    $(target).classList.add('active');
+    setTimeout(() => {
+      $(target).classList.remove('active');
+    }, alertShowTime);
+  }
```

Comment:
className 사용하실 거였으면 요게 더 나아보입니다!

```suggestion
  showAlert(targetClassName) {
    $(targetClassName).classList.add('active');
    setTimeout(() => {$(targetClassName).classList.remove('active')}, alertShowTime);
  }
```

Reply:
좋은 지적입니다 감사합니다:)